source/image-handler/image-request.js

```diff
@@ -448,6 +414,37 @@ class ImageRequest {
 
     return null;
   }
+
+  /**
+   * Return the output format depending on first four hex values of an image file.
+   * @param {Buffer} imageBuffer - Image buffer.
+   */
+  inferImageType(imageBuffer) {
+    switch (imageBuffer.toString('hex').substring(0, 8).toUpperCase()) {
```

Comment:
You could specify different arrays or a lookup object like this to avoid giant switch statements:
```js
// With arrays
const pngFirstBytes = ['89504E47'];
const jpegFirstBytes = ['FFD8FFDB', 'FFD8FFE0', 'FFD8FFEE', 'FFD8FFE1'];
const webpFirstBytes = ['52494646'];
const tiffFirstBytes = ['4D4D002A'];

const firstBytes = imageBuffer.toString('hex').substring(0, 8).toUpperCase();

if (pngFirstBytes.includes(firstBytes)) {
  return 'image/png';
} else if (jpegFirstBytes.includes(firstBytes)) {
  return 'image/jpeg';
} else if (webpFirstBytes.includes(firstBytes)) {
  return 'image/webp';
} else if (tiffFirstBytes.includes(firstBytes)) {
  return 'image/tiff';
} else {
  throw {
    status: 500,
      code: 'RequestTypeError',
      message: 'The file does not have an extension and the file type could not be inferred. Please ensure that your original image is of a supported file type (jpg, png, tiff, webp, svg). Refer to the documentation for additional guidance on forming image requests.'
  };
}


// With lookup object
const firstBytesMapping = {
  // PNG
  '89504E47': 'image/png',

  // JPEG
  'FFD8FFDB': 'image/jpeg',
  'FFD8FFE0': 'image/jpeg',
  'FFD8FFEE': 'image/jpeg',
  'FFD8FFE1': 'image/jpeg',
  
  // WEBP
  '52494646': 'image/webp',

  // TIFF
  '49492A00': 'image/tiff',
  '4D4D002A': 'image/tiff',
};

if (firstBytes in firstBytesMapping) {
   return firstBytesMapping[firstBytes];
} else {
  throw {
    status: 500,
      code: 'RequestTypeError',
      message: 'The file does not have an extension and the file type could not be inferred. Please ensure that your original image is of a supported file type (jpg, png, tiff, webp, svg). Refer to the documentation for additional guidance on forming image requests.'
  };
}
```

Reply:
generally a good thing, but we're doing some cpu/mem heavy image manipulations. This won't have a noticeable effect on the overall resource consumption. also, this stuff comes from `upstream`